### ISSUE TITLE
filterColumn coming as an "OR" condition instead of "AND"

### ISSUE THREAD

**Issue by sameerpanjwani** [yajra/laravel-datatables#57]
I don't think it was like this in one of your previous versions but not sure. Anyway, when I'm using fitlerColumn on the concat column and also doing a search on another individual column, then it's going as an "OR" condition instead of an "AND". However, if I do any other two column search, it goes as an "AND".

**Comment by yajra** [comment]
YES, filterColumn will always use "OR" as documented.

>Note: For global search, these methods are automaticaly converted to their "or" equivalents (if applicable, if not applicable, the column is not searched). If you do not want some column to be searchable in global search, set the javascript column flag searchable: false

**Comment by sameerpanjwani** [comment]
How do I make it use AND, not able to figure because other normal columns
search with AND.

**Comment by sameerpanjwani** [comment]
To be clear, am not talking about global search but column search.

**Comment by yajra** [comment]
use the extended Query Builder feature of Datatables. 

```php
$datatables =  Datatables::of($users);

if (...) {
$datatables->where('field','=',1);
}

return $datatables->make(true);
```

**Comment by sameerpanjwani** [comment]
Then your aggregate queries fail because the concat column doesn't exist.
Even if I use whereRaw, same problem. Only filter column works with concat .

Anyway I'll try to figure another way, thanks.

**Comment by yajra** [comment]
if your query is complex, I suggest you disable global search and write your query instead by using `filter` function.

```php
->filter(function($query){
   // your complex query here
   $query->where('field','=',1)
      ->whereRaw("concat(1,' ',2) like ?", [2]);
})
```

**Comment by yajra** [comment]
see this example for ref:

```php
    public function getCustomFilterData(Request $request)
    {
        $users = User::select(['id', 'name', 'email', 'created_at', 'updated_at']);

        return Datatables::of($users)
        ->filter(function ($query) use ($request) {
            if ($request->has('name')) {
                $query->where('name', 'like', "%{$request->get('name')}%");
            }

            if ($request->has('email')) {
                $query->where('email', 'like', "%{$request->get('email')}%");
            }
        })
        ->make(true);
    }
```

**Comment by sameerpanjwani** [comment]
Actually there is an issue I think with your script, a very weird issue. I just checked in your demo app, the multi-filter-select. filterColumn is working as an "AND" condition on the individual column fitler, even when you search in combination with the other columns.

However, just move the concat column to the end in the controller and blade file and it works as an "OR", why would that be?

```php
$users = User::select([   'name', 'email', 'created_at', 'updated_at',\DB::raw("CONCAT(users.id,'-',users.id) as user_id")]);
````

````
<thead>
		<tr>

			<th>Name</th>
			<th>Email</th>
			<th>Created At</th>
			<th>Updated At</th>
            <th>Id</th>
		</tr>
	</thead>
    <tfoot>
		<tr>

			<th>Name</th>
			<th>Email</th>
			<th>Created At</th>
			<th>Updated At</th>
            <th>Id</th>
		</tr>
	</tfoot>
````

````
 {data: 'name', name: 'name'},
            {data: 'email', name: 'email'},
            {data: 'created_at', name: 'created_at'},
            {data: 'updated_at', name: 'updated_at'},
            {data: 'user_id', name: 'user_id'}
````

**Comment by yajra** [comment]
I see your point now. For filterColumn on **global search** we should definitely use "OR" equivalent of the query. However, for **column search** we should use "AND" by default because it should match each row's search value.

**Comment by sameerpanjwani** [comment]
Yes and it for some reason works like an AND when you put it in the first
column but not otherwise.

**Comment by yajra** [comment]
It will work as AND if you did not use filterColumn but OR if you do.

**Comment by yajra** [comment]
Should be fixed on v5.2.1

**Comment by sameerpanjwani** [comment]
In your demo app, at least for me, it's showing AND for me in the query
with filter Column but only when it is the first column. Anyway if you fix
it, will be great. Thanks.
